File: evolution-of-todo/backend/app/schemas/todo.py
"""Todo schemas."""
from __future__ import annotations
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field

class TodoCreate(BaseModel):
    """Schema for creating a todo."""
    title: str = Field(min_length=1, max_length=200)
    description: Optional[str] = Field(default=None, max_length=1000)

class TodoUpdate(BaseModel):
    """Schema for updating a todo."""
    title: Optional[str] = Field(default=None, min_length=1, max_length=200)
    description: Optional[str] = Field(default=None, max_length=1000)
    completed: Optional[bool] = None

class TodoResponse(BaseModel):
    """Schema for todo response."""
    id: int
    user_id: int
    title: str
    description: Optional[str]
    completed: bool
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

File: evolution-of-todo/frontend-v2/src/types/todo.ts
export interface Todo {
    id: number;
    user_id: number;
    title: string;
    description: string | null;
    completed: boolean;
    created_at: string;
    updated_at: string;
}

export interface TodoCreate {
    title: string;
    description?: string;
}

export interface TodoUpdate {
    title?: string;
    description?: string;
    completed?: boolean;
}
